subsys/net/lib/http/http_client.c

```diff
@@ -34,7 +34,19 @@ static int sendall(int sock, const void *buf, size_t len)
 	while (len) {
 		ssize_t out_len = zsock_send(sock, buf, len, 0);
 
-		if (out_len < 0) {
+		if ((out_len == 0 && len != 0) || (out_len < 0 && errno == EAGAIN)) {
```

Comment:
`len != 0` check seems pointless here - if it wasn't true the loop wouldn't execute in the first place.